### DISCUSSION TITLE
Maintainer permissions

### DISCUSSION THREAD

**Discussion by ghostsquad** [go-task/task#724] [General]
@andreynering I'd like to do more in task, but I currently don't have very many permissions. Example, #723

Would it be useful to setup a slack or discord server maybe to improve communication?

**Comment by andreynering** [comment]
I can do that, but probably tomorrow night (got stuff to do by today).

We need some rules though, although unforced: please don't push new features or anything that needs enough discussion to master (or to the to-be-created `v4` branch). I want to review them, open a PR and put me as a reviewer 🙂. (At least for now, that can change in the future).

You're free to merge small bug fixes and documentation changes from others.

---

Any opinions on Slack vs. Discord?

---

FYI, I signed-up for [Carbon Ads](https://www.carbonads.net/open-source) and plan to maybe add it to https://taskfile.dev by tomorrow. If anyone is against this, this is the time to say, it still not a final decision...

We're are also on [OpenCollective](https://opencollective.com/task). If I get enough trust to the other collaborators, I'm open to share those donations. (Yes, it's not much, but that's not the point, it's a small "thank you" from some users 🙂).

**Reply by per1234** [reply]
On the subject of funding, have you considered signing up for [GitHub Sponsors](https://github.com/sponsors) @andreynering?

**Reply by andreynering** [reply]
@per1234 Yes, in fact I have signed up on the very first day of the announcement a long ago, but unfortunately they apparently do not support Brazilians yet. Never got any response from GitHub.

**Reply by ghostsquad** [reply]
Definitely will be careful and err on the side of talk more, change less. Backwards compatibility and forward flexibility is very important to me.

As for v4, I'm currently focused on `fn` for now (when I can find the time). Though, as I approach the release of an experimental version, I'll be thinking of how some of features can be ported to a v4 of `Task`.

**Reply by kerma** [reply]
There's a task channel in gophers Slack. Maybe use this one or create a private channel in Gophers?

**Reply by andreynering** [reply]
I just gave write access to all you three.

Also, @tylermmorton was added to the whole org like the others (he was only in the repository, by mistake)

I created a Discord server and linked in the website. Let's see how that works. Invite link: https://discord.gg/bVnczQBY

**Reply by per1234** [reply]
I see @andreynering is now in the GitHub Sponsors program!

https://github.com/sponsors/andreynering

I have settled on GitHub Sponsors as the most convenient way to donate to those who make valuable contributions to important open source software projects like Task, so I'm very glad this has happened.

**Reply by andreynering** [reply]
Thanks a lot @per1234!

Yes, it took some time, but GitHub Sponsors finally launched in Brazil!

https://github.blog/2022-06-01-github-sponsors-launches-in-brazil/

**Comment by ssbarnea** [comment]
Why not creating a matrix room instead? Discord is not really as accessible as matrix. Matrix is also bridged with irc, opening far more doors than discord.

**Reply by ghostsquad** [reply]
Benefits of discord, is that if you wanted to hop into a quick voice chat, you can very easily. What accessible concerns exist with Discord?

**Reply by ssbarnea** [reply]
You need a server and join it. Like Slack, you are expected to be joined into that server instance to interact with others. I already have 10 servers I joined and I already find that distracting. On matrix you have flatter experience as you join rooms, for which you do not need an invitation. In fact for matrix/irc, you can even access a room without having an account, which can be handy for people less used to it. Another benefit of matrix is that it also bridges with gitter. Sadly I did not see a working discord bridge.

Regarding accessibility, I found the UI of discord too animated/colored. Too many thing blinking around and distracting you.

Other than this, I observed that very few open-source projects are using discord, and even those that are using it, those servers do not seem as being very active. I cannot say the same about matrix, where I see a lot of activity. For example I am connected to >50 rooms, almost all opensource projects. Out of the 10 discord servers, only 5 are what I would open-source projects.

Not so sure about the voice chat, I used it only once on discord and on slack, and that was once a mistake that took a lot of time to figure out how to close it and the second time it was just for curiosity.

Matrix also has support for adhoc voice or video meeting via jitsi, directly at room level.

**Reply by andreynering** [reply]
Hi @ssbarnea, I appreciate your suggestions.

[Gitea](https://discord.com/invite/Gitea), [Go](https://discord.com/invite/golang) and [React](https://discord.com/invite/reactiflux) have Discord servers, just to have a few counter-examples for the "very few OSS projects use Discord" argument. I'm sure many others out there do as well, I see links to them often.

I was in doubt about Slack vs. Discord. Decided to try Discord. This is not final decision. But I really know lots of people that use Discord, and very few that have even mentioned Matrix (I have never used it myself). So I actually have the impression that Discord is actually more accessible in this regard.

And of course, there's GitHub as well, so issues and discussions (like this one!) are also ways to make contact for those who prefer it! 😉
